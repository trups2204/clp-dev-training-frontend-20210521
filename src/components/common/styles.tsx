import styled from 'styled-components';


export const GalleryContainer = styled.div`
padding-top:20px;
display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
`;

export const GalleryImage = styled.img`
vertical-align: middle;

`;

export const GalleryCaption = styled.span`
  margin: 0;
  padding: 1em;
  position: absolute;
  z-index: 1;
  bottom: 0;
  left: 0;
  width: 100%;
  max-height: 100%;
  overflow: auto;
  box-sizing: border-box;
  transition: transform 0.5s;
  transform: translateY(100%);
  background: rgba(0, 0, 0, 0.7);
  color: rgb(255, 255, 255);
 
`;
export const GalleryItem = styled.div`
position: relative;
overflow: hidden;
height:150px;
&:hover {
    ${GalleryCaption} {
        transform: translateY(0%);
    }
  }

`;
export const SpinnerPanel = styled.div`
position: fixed;
left: 50%;
top: 50%;
width: 100%;
height: 100%;
z-index: 9999;
  @keyframes spin {
    from {
      stroke-dashoffset: 360;
    }
    to {
      stroke-dashoffset: 40;
    }
  }

`;

export const SpinnerSvg= styled.svg`
  transform: rotate(-90deg);
  width: 60px;
  height: 60px;
`;

export const SvgCircle = styled.circle`
stroke-linecap: round;
stroke-dasharray: 360;
stroke-width: 6;
stroke: gray;
fill: transparent;
animation: spin .6s ease-in-out 0s normal infinite;
`;

export const ScreenReaderOnly = styled.p`
  position:absolute;
  left:-10000px;
  top:auto;
  width:1px;
  height:1px;
  overflow:hidden;
`;