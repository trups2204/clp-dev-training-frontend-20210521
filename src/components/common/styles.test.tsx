import React from 'react';
import renderer from 'react-test-renderer';
import * as styles from './styles';

describe('styles',() =>{

    it('should render GalleryContainer',() =>{
        const component = <styles.GalleryContainer/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('should render GalleryImage',() =>{
        const component = <styles.GalleryImage/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('should render GalleryCaption',() =>{
        const component = <styles.GalleryCaption/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('should render GalleryItem',() =>{
        const component = <styles.GalleryItem/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
    it('should render SpinnerPanel',() =>{
        const component = <styles.SpinnerPanel/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
    it('should render SpinnerSvg',() =>{
        const component = <styles.SpinnerSvg/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
    it('should render SvgCircle',() =>{
        const component = <styles.SvgCircle/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
    it('should render ScreenReaderOnly',() =>{
        const component = <styles.ScreenReaderOnly/>;
        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });

    
})