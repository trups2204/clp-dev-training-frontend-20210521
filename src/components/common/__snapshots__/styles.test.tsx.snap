// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles should render GalleryCaption 1`] = `
<span
  className="sc-dlnjwi eWwnEx"
/>
`;

exports[`styles should render GalleryContainer 1`] = `
<div
  className="sc-bdnxRM iwaRrM"
/>
`;

exports[`styles should render GalleryImage 1`] = `
<img
  className="sc-gtsrHT imHcTG"
/>
`;

exports[`styles should render GalleryItem 1`] = `
<div
  className="sc-hKFxyN kXSnkB"
/>
`;

exports[`styles should render ScreenReaderOnly 1`] = `
<p
  className="sc-iCoGMd boJwNj"
/>
`;

exports[`styles should render SpinnerPanel 1`] = `
<div
  className="sc-eCApnc lnjSS"
/>
`;

exports[`styles should render SpinnerSvg 1`] = `
<svg
  className="sc-jSFjdj brjlgd"
/>
`;

exports[`styles should render SvgCircle 1`] = `
<circle
  className="sc-gKAaRy jgrAZo"
/>
`;
